mico-core/src/main/java/io/github/ust/mico/core/web/ApplicationController.java

```diff
@@ -155,21 +176,24 @@
             // application already deleted
             return ResponseEntity.noContent().build();
         }
+        
         applicationOptional.map(application -> {
-            if (application.getDeploymentInfo() == null || application.getDeploymentInfo().getServiceDeploymentInfos() != null) {
-                return application; // TODO better deployment detection
+            try {
+                if (micoKubernetesClient.isApplicationDeployed(application)) {
+                    throw new ResponseStatusException(HttpStatus.CONFLICT, "Application is currently deployed!");
+                } else {
+                    return application;
+                }
+            } catch (KubernetesResourceException e) {
+                log.error(e.getMessage(), e);
+                // TODO: Check whether CONFLICT is the best status option here
```

Comment:
I think 409 Conflict is correct.
Do really want to have this almost useless try-catch?

Reply:
Try-catch removed by replacing lambda expression with simple if-else-statement and throws declaration.